### PR TITLE
Fix/item render ready

### PR DESCRIPTION
**Pull request by krampstudio**
https://github.com/oat-sa/extension-tao-itemqti-pic/pull/17 is needed to test with PIC.

It fixes the readiness of the item: we wait all elements' postRendering to be finished before continuing

Based on task TAO-964, but it may affect all QTI elements and interactions during preview, authoring and delivery.
Exhaustive testing will be made by QA, PR testing can be made on a few interactions (at least one block interaction, one inline interaction, the extended text, one graphical interaction, a PCI and a PIC)

### CODE REVIEW

File: views/js/qtiCommonRenderer/renderers/Math.js [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
-        render : function(math, data){
-            if(typeof(MathJax) !== 'undefined' && MathJax){
-                _.delay(function(){//defer execution fix some rendering issue in chrome
-                     MathJax.Hub.Queue(["Typeset", MathJax.Hub, containerHelper.get(math).parent()[0]]);
-                },60);
-            }
+        render : function render (math, data){
+            return new Promise(function(resolve, reject){
+                if(typeof(MathJax) !== 'undefined' && MathJax){
+                    _.delay(function(){//defer execution fix some rendering issue in chrome
```
Comment (ssipasseuth) [inline views/js/qtiCommonRenderer/renderers/Math.js]: would be nice to replace the delay with a proper way to listen when the mathjax typesetting is ready ...
like that ? http://docs.mathjax.org/en/latest/signals.html

**Comment by ssipasseuth** [comment]
sorry, had to assign it back to you because of missing time

File: views/js/qtiCommonRenderer/renderers/Img.js [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
+    'tpl!taoQtiItem/qtiCommonRenderer/tpl/img',
+    'taoQtiItem/qtiCommonRenderer/helpers/container'
+], function(tpl, containerHelper){
+    'use strict';
+
     return {
         qtiClass : 'img',
         template : tpl,
-        getContainer : containerHelper.get
+        getContainer : containerHelper.get,
```
Comment (jsconan) [inline views/js/qtiCommonRenderer/renderers/Img.js]: Extra comma can introduce issue in some javascript engines, like IE

File: views/js/qtiCommonRenderer/renderers/PortableInfoControl.js [hunk 3]
```diff
@@ -21,12 +21,13 @@
  * Portable Info Control Common Renderer
  */
 define([
+    'core/promise',
     'tpl!taoQtiItem/qtiCommonRenderer/tpl/infoControl',
     'taoQtiItem/qtiCommonRenderer/helpers/container',
     'taoQtiItem/qtiCommonRenderer/helpers/PortableElement',
     'qtiInfoControlContext',
     'taoQtiItem/qtiItem/helper/util',
```
Comment (jsconan) [inline views/js/qtiCommonRenderer/renderers/PortableInfoControl.js]: Extra comma can introduce issue in some javascript engines, like IE

File: views/js/qtiCommonRenderer/renderers/interactions/GraphicAssociateInteraction.js [hunk 4]
```diff
@@ -24,14 +24,14 @@ define([
...
     'lodash',
     'i18n',
-    'raphael',
-    'scale.raphael',
+    'core/promise',
     'tpl!taoQtiItem/qtiCommonRenderer/tpl/interactions/graphicAssociateInteraction',
     'taoQtiItem/qtiCommonRenderer/helpers/Graphic',
     'taoQtiItem/qtiCommonRenderer/helpers/PciResponse',
     'taoQtiItem/qtiCommonRenderer/helpers/container',
     'taoQtiItem/qtiCommonRenderer/helpers/instructions/instructionManager',
```
Comment (jsconan) [inline views/js/qtiCommonRenderer/renderers/interactions/GraphicAssociateInteraction.js]: Extra comma can introduce issue in some javascript engines, like IE

File: views/js/qtiRunner/core/Renderer.js [hunk 5]
```diff
@@ -27,10 +27,11 @@ define([
     'lodash',
     'jquery',
     'handlebars',
+    'core/promise',
     'taoQtiItem/qtiItem/core/Element',
     'taoQtiItem/qtiItem/helper/interactionHelper',
     'taoQtiItem/runner/themes/loader',
```
Comment (jsconan) [inline views/js/qtiRunner/core/Renderer.js]: Extra comma can introduce issue in some javascript engines, like IE

File: views/js/runner/provider/qti.js [hunk 6]
```diff
@@ -23,15 +23,18 @@
...
     'jquery',
     'lodash',
-    'async',
+    'context',
+    'core/promise',
     'taoQtiItem/qtiItem/core/Loader',
     'taoQtiItem/qtiItem/core/Element',
     'taoQtiItem/qtiCommonRenderer/renderers/Renderer',
     'taoQtiItem/runner/provider/manager/picManager',
     'taoItems/assets/manager',
```
Comment (jsconan) [inline views/js/runner/provider/qti.js]: Extra comma can introduce issue in some javascript engines, like IE

**Pull request merged**